lib/contentful/client.rb

```diff
@@ -184,10 +192,83 @@ def base_url
       "http#{configuration[:secure] ? 's' : ''}://#{configuration[:api_url]}/spaces/#{configuration[:space]}"
     end
 
+    # Returns the formatted part of the X-Contentful-User-Agent header
+    # @private
+    def format_user_agent_header(key, values)
+      header = "#{key} #{values[:name]}"
+      header = "#{header}/#{values[:version]}" if values[:version]
+      "#{header};"
+    end
+
+    # Returns the X-Contentful-User-Agent sdk data
+    # @private
+    def sdk_info
+      {
+        name: 'contentful.rb',
+        version: ::Contentful::VERSION
+      }
+    end
+
+    # Returns the X-Contentful-User-Agent app data
+    # @private
+    def app_info
+      {
+        name: configuration[:application_name],
+        version: configuration[:application_version]
+      }
+    end
+
+    # Returns the X-Contentful-User-Agent integration data
+    # @private
+    def integration_info
+      {
+        name: configuration[:integration_name],
+        version: configuration[:integration_version]
+      }
+    end
+
+    # Returns the X-Contentful-User-Agent platform data
+    # @private
+    def platform_info
+      {
+        name: 'ruby',
+        version: RUBY_VERSION
+      }
+    end
+
+    # Returns the X-Contentful-User-Agent os data
+    # @private
+    def os_info
+      {
+        name: Gem::Platform.local.os,
```

Comment:
I don't think the output from this function (https://github.com/rubygems/rubygems/blob/dca0a654329aa24d2c7435a3fa486769dd89b7b4/lib/rubygems/platform.rb#L79) doesn't match our spec, you'll have to normalise it.